### PR TITLE
Add coteacher deadline

### LINKED ISSUE #2371: Coteacher Deadline
There's currently no coteacher deadline (it depends on Teacher/MainPage).
This could be problematic for programs that need to finalize their teacher lists by a certain deadline for training, printing, etc.

Requested by Stanford.

### PR DESCRIPTION
**Pull request by willgearty**
Changes the deadline dependency of the coteachers page from `/Teacher/MainPage` to `/Teacher/Classes/Coteachers` so the coteachers page can be closed independently.

Fixes #2371.

### CODE REVIEW

File: esp/esp/users/migrations/0015_auto_20180709_1328.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```
Comment (kkbrum) [inline esp/esp/users/migrations/0015_auto_20180709_1328.py]: For my understanding-- what is happening with these migration files?

Comment (willgearty) [reply esp/esp/users/migrations/0015_auto_20180709_1328.py]: The migrations change the way the database is set up. They can add fields, remove fields, or change fields. In this case, this is modifying the potential options for the permission field.

Comment (willgearty) [reply esp/esp/users/migrations/0015_auto_20180709_1328.py]: Any time you make a modification to the database models, you need to create a migration for it. `fab manage:makemigrations` makes the migration files for you, which you then need to `git add`. To then update the database of your own dev server, you do `fab manage:migrate`.

Comment (kkbrum) [reply esp/esp/users/migrations/0015_auto_20180709_1328.py]: oh cool! thanks

**Pull request merged**